ccswbs/gus
# Add Hero video option for Program pages

# Summary of Changes

## Frontend
- Feature branch URL: https://tqtest.gatsbyjs.io/
- New plugin: [react-player](https://github.com/cookpete/react-player)
- New component: [heroVideo.js](https://github.com/ccswbs/gus/pull/42/files#diff-b5e9d505fc92fc31d61e5165fd5a3e098139c5c687df4efc285e462a9c8b786a)
- New stylesheet: [heroVideo.css](https://github.com/ccswbs/gus/pull/42/files#diff-55017c62566ceca3c9bdb5f5c97bd3141a873461ab88094bb46c32e8803d91b3)
- Program page checks if Hero component is an image or video 
- Null checks for video width and height added to `gatsby-node.js`

## Backend
- Multidev URL: https://herovid-chug.pantheonsite.io/
- Video width and height fields added to Remote Video media type
- Remote Video added as option for Program page feature area
- Improvements to Media views (filter by video url and tags)
- Removed obsolete Video custom content type

## Test Plan
- Go to https://tqtest.gatsbyjs.io/programs/associate-diploma-in-testing-technology-and-ridiculously-long-program-names-that-wrap
- Verify presence of placeholder until video loads 
- Verify play/pause button works and is keyboard-accessible
- Resize window and verify video scales in a responsive manner
- Go to https://tqtest.gatsbyjs.io/programs/master-of-data-science
- Verify Download Transcript icon/link appears and works
- Go to https://herovid-chug.pantheonsite.io/
- Create a new remote video
- Verify the width and height fields are automatically populated after you click "Save"
- Edit an existing program, preferably one that doesn't already have a feature image
- Click "Add media"
- Click the "Remote video" tab
- Type _vimeo_ in the "Video URL" field and verify only Vimeo videos are returned
- Type _you_ in the same field and verify YouTube videos are returned instead
